framework/src/modules/token/cc_commands/cc_transfer.ts

```diff
@@ -49,104 +46,154 @@ export class CCTransferCommand extends BaseCCCommand {
 		return CROSS_CHAIN_COMMAND_NAME_TRANSFER;
 	}
 
-	public addDependencies(interoperabilityMethod: InteroperabilityMethod) {
-		this._interopMethod = interoperabilityMethod;
+	public init(args: { ownChainID: Buffer }): void {
+		this._ownChainID = args.ownChainID;
 	}
 
-	public init(args: { minBalances: MinBalance[]; supportedTokenIDs: Buffer[] }): void {
-		this._supportedTokenIDs = args.supportedTokenIDs;
-		// this._minBalances = args.minBalances;
-	}
-
-	public async execute(ctx: CCCommandExecuteContext): Promise<void> {
+	public async verify(ctx: CCCommandExecuteContext): Promise<void> {
 		const { ccm } = ctx;
 		const methodContext = ctx.getMethodContext();
-		const { id: ownChainID } = await this._interopMethod.getOwnChainAccount(methodContext);
+		const ownChainID = this._ownChainID;
 		let params: CCTransferMessageParams;
-		let tokenChainID;
-		// let tokenLocalID;
+		const mainChainID = this._tokenMethod.getMainchainTokenID();
+		let tokenChainID: Buffer;
+		const { sendingChainID } = ccm;
+
 		try {
 			params = codec.decode<CCTransferMessageParams>(crossChainTransferMessageParams, ccm.params);
 			validator.validate(crossChainTransferMessageParams, params);
+			const { tokenID, amount } = params;
+			[tokenChainID] = splitTokenID(tokenID);
+
+			if (
+				!tokenChainID.equals(ownChainID) &&
+				!tokenChainID.equals(sendingChainID) &&
+				!tokenChainID.equals(mainChainID)
+			) {
+				throw new Error(
+					'Token must be native to either the sending or the receiving chain or the mainchain.',
+				);
+			}
 
-			[tokenChainID] = splitTokenID(params.tokenID);
 			if (tokenChainID.equals(ownChainID)) {
+				// TODO: recheck if this is correct
+				// if escrowStore(sendingChainID, tokenID) does not exist
+				// or escrowAmount(sendingChainID, tokenID) < amount:
+				// raise Exception("Insufficient balance in escrow account.")
 				const escrowedAmount = await this._tokenMethod.getEscrowedAmount(
 					methodContext,
-					ccm.sendingChainID,
-					params.tokenID,
+					sendingChainID,
+					tokenID,
 				);
-				if (params.amount > escrowedAmount) {
-					throw new Error(
-						`Amount ${params.amount.toString()} is not sufficient for ${escrowedAmount.toString()}`,
-					);
+
+				if (escrowedAmount < amount) {
+					throw new Error('Insufficient balance in escrow account.');
 				}
 			}
 		} catch (error) {
 			ctx.logger.debug({ err: error as Error }, 'Error verifying the params.');
-			if (ccm.status === CCM_STATUS_OK && ccm.fee >= MIN_RETURN_FEE * ctx.ccmSize) {
-				await this._interopMethod.error(methodContext, ccm, CCM_STATUS_PROTOCOL_VIOLATION);
+
+			if (!(error instanceof NotFoundError)) {
```

Comment:
from above, only `getEscrowedAmount` will throw `NotFoundError` and in this case it should be error.
I think above, probably it will make it more readable to make the scope of `try-catch` smaller

Reply:
here the logic is if it doesn't exist, you need to reject. (also, it should be reflected in the test as well)